homeassistant/components/yeelight/config_flow.py

```diff
@@ -38,30 +39,32 @@ def async_get_options_flow(config_entry):
 
     def __init__(self):
         """Initialize the config flow."""
-        self._capabilities = None
         self._discovered_devices = {}
+        self._host = None
+        self._capabilities = None
 
     async def async_step_user(self, user_input=None):
         """Handle the initial step."""
         errors = {}
         if user_input is not None:
             if user_input.get(CONF_HOST):
                 try:
-                    await self._async_try_connect(user_input[CONF_HOST])
-                    return self.async_create_entry(
-                        title=self._async_default_name(),
-                        data=user_input,
-                    )
+                    self._host = user_input[CONF_HOST]
+                    await self._async_try_connect(self._host)
+                    return await self.async_step_name()
```

Comment:
I think that we should consider dropping the name altogether and just use the host as name

Reply:
I've looked at the name that we suggest now, those names are not very useful. We can just set it o IP and don't use names at all. People can pick their own names since there is a unique ID in the entity registry.

Reply:
Before this PR `name` is only used for backwards compatibility to YAML configuration. What about keep the current default name and fallback to IP address when we cannot get capabilities from the device? Discovered devices don't necessary have static IP address so it's not possible to use IP address as default name for all devices.

Reply:
Sounds good.